File: CCC/Golf.cpp
// CCC '00 S4 - Golf

#include<cstdio>
#include<algorithm>
#include<cstring>

int dis, n, dp[5300];
int main(){
    scanf("%d%d", &dis, &n);

    memset(dp, 0x3f3f3f3f, sizeof(dp));
    dp[0] = 0;
    for (int i=0; i<n; i++) {
        int w; scanf("%d", &w);
        for (int j=w; j<=dis; j++){
            dp[j] = std::min(dp[j], dp[j-w]+1);
        }
    }
    if (dp[dis] == 0x3f3f3f3f)
        printf("Roberta acknowledges defeat.\n");
    else printf("Roberta wins in %d strokes.", dp[dis]);

    return 0;
}

File: DMOJ/CCC/Golf.py
import sys
from itertools import combinations
input = sys.stdin.readline


dis = int(input())

c = int(input())
clubs=[]
for i in range(c):
    clubs.append(int(input()))

n = 35
for i in range(0,c+1):
    for comb in list(combinations(clubs,i)):
        #print(comb, sum(comb), len(comb))
        s = sum(comb)
        if s==dis:
            n = min(n,len(comb))


if n==35: print("Roberta acknowledges defeat.")
else: print(f"Roberta wins in {n} strokes.")